deepforest/visualize.py

```diff
@@ -257,37 +257,225 @@ def label_to_color(label):
+        polygons = [np.array(p).round().astype(np.int32) for p in polygons]
+        masks = [sv.polygon_to_mask(p, (width, height)) for p in polygons]
+        masks = np.stack(masks)
+
+        detections = sv.Detections(
+            xyxy=boxes,
+            mask=masks,
+            class_id=labels,
+            confidence=scores,
+            data={"class_name": [class_name[class_id] for class_id in labels]})
+
+    elif geom_type == "point":
+        points = df.geometry.apply(lambda x: (x.x, x.y)).values
+        points = np.stack(points)
+        points = np.expand_dims(points, axis=1)
+
+        label_mapping = {label: idx for idx, label in enumerate(df['label'].unique())}
+
+        # Extract labels as a numpy array
+        labels = df['label'].map(label_mapping).values.astype(int)
 
-    label_mapping = {label: idx for idx, label in enumerate(df['label'].unique())}
+        # Extract scores as a numpy array
+        scores = np.array(df['score'].tolist())
+        scores = np.expand_dims(np.stack(scores), 1)
 
-    # Extract labels as a numpy array
-    labels = df['label'].map(label_mapping).values.astype(int)
+        # Create a reverse mapping from integer to string labels
+        class_name = {v: k for k, v in label_mapping.items()}
 
-    # Extract scores as a numpy array
-    scores = np.array(df['score'].tolist())
-    # Create a reverse mapping from integer to string labels
-    class_name = {v: k for k, v in label_mapping.items()}
+        detections = sv.KeyPoints(
+            xy=points,
+            class_id=labels,
+            confidence=scores,
+            data={"class_name": [class_name[class_id] for class_id in labels]})
 
-    return sv.Detections(
-        xyxy=boxes,
-        class_id=labels,
-        confidence=scores,
-        data={"class_name": [class_name[class_id] for class_id in labels]})
+    return detections
+
+
+def plot_results(results,
+                 ground_truth=None,
+                 savedir=None,
+                 height=None,
+                 width=None,
+                 results_color=[245, 135, 66],
+                 ground_truth_color=[0, 165, 255],
+                 thickness=2,
+                 basename=None,
+                 radius=3):
+    """Plot the prediction results.
+
+    Args:
+        df: a pandas dataframe with prediction results
+        ground_truth: an optional pandas dataframe with ground truth annotations
+        savedir: optional path to save the figure. If None (default), the figure will be interactively plotted.
+        height: height of the image in pixels. Required if the geometry type is 'polygon'.
+        width: width of the image in pixels. Required if the geometry type is 'polygon'.
+        basename: optional basename for the saved figure. If None (default), the basename will be extracted from the image path.
+        results_color (list): color of the results annotations as a tuple of RGB color, e.g. orange annotations is [245, 135, 66]
```

Comment:
Missing ground_truth_color, thickness, and radius. Keep arguments in same order as function definition